plugins/events-correlation-engine/src/main/java/org/opensearch/plugin/correlation/rules/action/IndexCorrelationRuleAction.java

```diff
@@ -0,0 +1,21 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.plugin.correlation.rules.action;
+
+import org.opensearch.action.ActionType;
+
+public class IndexCorrelationRuleAction extends ActionType<IndexCorrelationRuleResponse> {
```

Comment:
We will also need to corresponding ``GetCorrelationRuleAction`` along with its REST and Transport implementations?

Reply:
will add more actions with individual prs.